### PR TITLE
Add Windows specific exec root for plugins.

### LINKED ISSUE #30572: Windows: dockerd.exe creates C:\run\docker\plugins folder
**Description**

After starting the Windows Docker engine a folder `C:\run\docker\plugins` exists. For Windows I think this should be in `C:\ProgramData\docker\plugins`. This folder also exists.

**Steps to reproduce the issue:**
1. stop-service docker
2. rm C:\tmp
3. start-service docker
4. dir C:\

**Describe the results you received:**

Folder C:\run\docker\plugins exists

**Describe the results you expected:**

This folder should not exist at toplevel.

**Additional information you deem important (e.g. issue happens only occasionally):**

It seems there are several places in the codebase that uses `"/run/docker/plugins"`.
I just found out that this also happens with 1.13.0.

**Output of `docker version`:**

```
Client:
 Version:      1.13.1-rc1
 API version:  1.25
 Go version:   go1.7.4
 Git commit:   2527cfc
 Built:        Sat Jan 28 00:43:00 2017
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.1-rc1
 API version:  1.25 (minimum version 1.24)
 Go version:   go1.7.4
 Git commit:   2527cfc
 Built:        Sat Jan 28 00:43:00 2017
 OS/Arch:      windows/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 21
 Running: 0
 Paused: 0
 Stopped: 21
Images: 15
Server Version: 1.13.1-rc1
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.693.amd64fre.rs1_release.161220-1747)
Operating System: Windows Server 2016 Standard Evaluation
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 2 GiB
Name: vagrant-2016
ID: M766:Y53P:D77L:DK43:TASA:SFRA:HMP6:6ICO:X3HL:DZBK:RJMN:FMFQ
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Azure VM

### PR DESCRIPTION
**Pull request by anusha-ragunathan**
Fixes #30572

Signed-off-by: Anusha Ragunathan <anusha.ragunathan@docker.com>

### CODE REVIEW

File: daemon/daemon_windows.go [hunk 1]
```diff
@@ -37,6 +38,8 @@ const (
 	windowsMinCPUCount   = 1
 )
 
+var platformExecRoot = filepath.Join(os.Getenv("programdata"), "docker", "plugins")
```
Comment (jhowardmsft) [inline daemon/daemon_windows.go]: This should be following the config.Root to allow two daemons to run side-by-side with out clashing directories, or when -g is specified on the daemon startup.

**Comment by LK4D4** [comment]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
Should it be for 1.13.1 ? 👼

**Comment by thaJeztah** [comment]
Discussed with @anusha-ragunathan  - plugins are not supported yet on Windows, so not necessary for 1.13.1

**Comment by tonistiigi** [comment]
nit: If windows doesn't have a concept of `ExecRoot` it should probably just completely ignore the property instead of requiring it to be a directory that already exists.

**Pull request merged**